### PR TITLE
Add shipping method channel listing

### PR DESCRIPTION
**Pull request by d-wysocki**
I want to merge this change because...I want to move everything related with shipping method price to `ShippingMethodChannelListing`

# Impact

* [X] New migrations
* [X] New/Updated API fields or mutations
* [X] Deprecated API fields or mutations
* [X] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [X] Privileged queries and mutations are guarded by proper permission checks
* [X] Database queries are optimized and the number of queries is constant
* [X] Database migration files are up to date
* [X] The changes are tested
* [X] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/order/utils.py [hunk 1]
```diff
@@ -307,7 +307,7 @@ def sum_order_totals(qs):
 
 def get_valid_shipping_methods_for_order(order: Order):
     return ShippingMethod.objects.applicable_shipping_methods_for_instance(
-        order, price=order.get_subtotal().gross
+        order, channel=order.channel, price=order.get_subtotal().gross
```
Comment (fowczarek) [inline saleor/order/utils.py:310]: Maby better to pass only `channel id` instead of getting channel object from the database? 

```suggestion
        order, channel_id=order.channel_id, price=order.get_subtotal().gross
```

File: saleor/shipping/models.py [hunk 2]
```diff
@@ -209,6 +227,41 @@ def get_total(self):
...
+        on_delete=models.CASCADE,
+    )
+    channel = models.ForeignKey(
+        Channel,
+        null=False,
+        blank=False,
+        related_name="shipping_method_listing",
+        on_delete=models.CASCADE,
+    )
+    price = models.DecimalField(
```
Comment (fowczarek) [inline saleor/shipping/models.py:245]: Why `price`, `min_value` and `max_value` are still on `ShippingMethod` model?

File: saleor/tests/fixtures.py [hunk 3]
```diff
@@ -258,14 +259,18 @@ def checkout_with_shipping_required(checkout_with_item, product):
...
-    return ShippingMethod.objects.create(
+def other_shipping_method(shipping_zone, channel_USD):
+    method = ShippingMethod.objects.create(
         name="DPD",
         minimum_order_price=Money(0, "USD"),
         type=ShippingMethodType.PRICE_BASED,
         price=Money(9, "USD"),
         shipping_zone=shipping_zone,
     )
+    ShippingMethodChannelListing.objects.create(
```
Comment (fowczarek) [inline saleor/tests/fixtures.py:270]: Why you don't move price to the channel listing?

File: saleor/tests/fixtures.py [hunk 4]
```diff
@@ -505,67 +510,85 @@ def staff_users(staff_user):
...
-    shipping_zone.shipping_methods.create(
+    method = shipping_zone.shipping_methods.create(
         name="DHL",
         minimum_order_price=Money(0, "USD"),
         type=ShippingMethodType.PRICE_BASED,
         price=Money(10, "USD"),
         shipping_zone=shipping_zone,
     )
+    ShippingMethodChannelListing.objects.create(
+        channel=channel_USD, shipping_method=method
```
Comment (fowczarek) [inline saleor/tests/fixtures.py:525]: Why you don't move price to the channel listing?

File: saleor/shipping/models.py (hunk 2 above)
Comment (fowczarek) [reply saleor/shipping/models.py:245]: Instead of `price` as the decimal field, you should move `price` as MoneyField with require `currency`  and `price_amount`.

Comment (fowczarek) [reply saleor/shipping/models.py:245]: Same situation with `minimum_order_price` and `maximum_order_price` should be `MoneyField`

File: saleor/tests/fixtures.py [hunk 5]
```diff
@@ -505,67 +510,85 @@ def staff_users(staff_user):
...
         type=ShippingMethodType.PRICE_BASED,
         price=Money(10, "USD"),
         shipping_zone=shipping_zone,
     )
+    ShippingMethodChannelListing.objects.bulk_create(
+        [
+            ShippingMethodChannelListing(channel=channel_USD, shipping_method=method),
+            ShippingMethodChannelListing(
+                channel=channel_USD, shipping_method=second_method
+            ),
```
Comment (fowczarek) [inline saleor/tests/fixtures.py:557]: Why you don't move price to the channel listing?

File: saleor/tests/fixtures.py [hunk 6]
```diff
@@ -505,67 +510,85 @@ def staff_users(staff_user):
...
-    shipping_zone.shipping_methods.create(
+    method = shipping_zone.shipping_methods.create(
         name="DHL",
         minimum_order_price=Money(0, "USD"),
         type=ShippingMethodType.PRICE_BASED,
         price=Money(10, "USD"),
         shipping_zone=shipping_zone,
     )
+    ShippingMethodChannelListing.objects.create(
+        channel=channel_USD, shipping_method=method
```
Comment (fowczarek) [inline saleor/tests/fixtures.py:574]: Why you don't move price to the channel listing?

File: saleor/tests/fixtures.py [hunk 7]
```diff
@@ -505,67 +510,85 @@ def staff_users(staff_user):
...
+def shipping_method(shipping_zone, channel_USD):
+    method = ShippingMethod.objects.create(
         name="DHL",
         minimum_order_price=Money(0, "USD"),
         type=ShippingMethodType.PRICE_BASED,
         price=Money(10, "USD"),
         shipping_zone=shipping_zone,
     )
+    ShippingMethodChannelListing.objects.create(
+        shipping_method=method, channel=channel_USD
```
Comment (fowczarek) [inline saleor/tests/fixtures.py:589]: Why you don't move price to the channel listing?

**Review by fowczarek — CHANGES_REQUESTED** [review]

**Review by fowczarek — CHANGES_REQUESTED** [review]
You add an empty test file, It should be removed. 
image

File: saleor/plugins/manager.py [hunk 8]
```diff
@@ -141,10 +143,15 @@ def calculate_checkout_shipping(
...
+    def calculate_order_shipping(self, order: "Order", channel_id) -> TaxedMoney:
         if not order.shipping_method:
             return zero_taxed_money(order.currency)
-        shipping_price = order.shipping_method.price
+        try:
+            shipping_price = order.shipping_method.channel_listing.get(
+                channel_id=channel_id
+            ).price
+        except ShippingMethodChannelListing.DoesNotExist:
+            shipping_price = None
```
Comment (IKarbowiak) [inline saleor/plugins/manager.py:154]: The same here 👮‍♀️

**Review by IKarbowiak — APPROVED** [review]

File: saleor/plugins/manager.py [hunk 9]
```diff
@@ -141,10 +143,15 @@ def calculate_checkout_shipping(
             "calculate_checkout_shipping", default_value, checkout, lines, discounts
         )
 
-    def calculate_order_shipping(self, order: "Order") -> TaxedMoney:
+    def calculate_order_shipping(self, order: "Order", channel_id) -> TaxedMoney:
```
[the remark this answers is not in the document]
Comment (fowczarek) [reply saleor/plugins/manager.py:146]: Maybe we should get channel_id form order instead of passing it as param? 
In which situation we should have different channel id than `order.channel`

**Review by fowczarek — COMMENTED** [review]
You should add a test for trying to add new ShippippingPriceChannelListing with `None` as price. `None` as price should be able only when `ShippippingPriceChannelListing` has set price.

**Review by fowczarek — APPROVED** [review]
:rocket:

**Review by IKarbowiak — APPROVED** [review]
